.github/workflows/test.yml

```diff
@@ -21,3 +21,22 @@ jobs:
 
       - name: Test packages
         run: yarn test
+
+  e2e-tests:
+    name: Run e2e tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2.1.4
+        with:
+          node-version: 14
+
+      - run: |
+          yarn install --frozen-lockfile
+          yarn cypress cache path
+          yarn cypress cache list
+          yarn cypress verify
+          yarn cypress info
+          yarn e2e
```

Comment:
Have to figure out a way to:

1. Export code coverage information to PR summary
2. Retain artifacts generated by Cypress (screenshots, video)
3. Cache `node_modules` for performance, could potentially be reused for all workflows and jobs
4. Whether to run a browser matrix here (with no mobile support) or use BrowserStack

Reply:
Can't answer any of these, would require researching same as for you.

However, because a preview of Surge allows to deploy multiple jobs on one PR, maybe it can be used to host both Cypress report and generated assets?

For example, on one of your apps we do store test assets as artifacts on GitLab (maybe [GitHub has similar approach](https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts)?), but for the main branch it's stored on GitLab Pages.

Also, do not worry about caching `node_modules` for now. I agree this could be handy, however can be a separate thing.

Reply:
Yea we can use Cypress' action, seems pretty good.
- https://github.com/cypress-io/github-action
- https://www.cypress.io/blog/2019/11/20/drastically-simplify-your-testing-with-cypress-github-action

I'll try it out later.